cylc/flow/workflow_files.py

```diff
@@ -1246,14 +1247,22 @@ def infer_latest_run(
             the input arg path.
         reg: The workflow name (including the numbered run if applicable).
 
-    Raises WorkflowFilesError if the runN symlink is not valid.
+    Raises:
+        - WorkflowFilesError if the runN symlink is not valid.
+        - UserInputError if the path does not exist.
     """
     cylc_run_dir = get_cylc_run_dir()
     try:
         reg = path.relative_to(cylc_run_dir)
     except ValueError:
         raise ValueError(f"{path} is not in the cylc-run directory")
+    if not path.exists():
+        raise UserInputError(f'Path does not exist: {path}')
     if path.name == WorkflowFiles.RUN_N:
+        LOG.warning(
+            f"The use of {WorkflowFiles.RUN_N} in the Workflow ID is not"
+            f" necessary."
```

Comment:
Not a biggie, but this ought to be an infrequently-seen warning, so I think we could afford to be more explicit, and take the opportunity to explain what the `runN` symlink is actually for:
```suggestion
            f"Explicit use of {WorkflowFiles.RUN_N} in the Workflow ID is not"
            f" necessary. It is used automatically to select the latest run number."
```